File: crashlytics-slf4j/src/main/java/org/slf4j/impl/StaticLoggerBinder.java
package org.slf4j.impl;

import org.slf4j.ILoggerFactory;
import org.slf4j.spi.LoggerFactoryBinder;

import io.leomo.android.slf4j.crashlytics.CrashlyticsLoggerFactory;

public class StaticLoggerBinder implements LoggerFactoryBinder {

    private static final CrashlyticsLoggerFactory factory = new CrashlyticsLoggerFactory();
    private static final StaticLoggerBinder binder = new StaticLoggerBinder();

    private StaticLoggerBinder() {
        // Nothing to do...
    }

    /* ====================================================================== */

    @Override
    public ILoggerFactory getLoggerFactory() {
        return factory;
    }

    @Override
    public String getLoggerFactoryClassStr() {
        return factory.getClass().getName();
    }

    /* ====================================================================== */

    public static StaticLoggerBinder getSingleton() {
        return binder;
    }
}

File: crashlytics-slf4j/src/main/java/org/slf4j/impl/StaticLoggerBinder.kt
package org.slf4j.impl

import org.slf4j.spi.LoggerFactoryBinder
import org.slf4j.ILoggerFactory
import de.lemona.android.crashlytics.CrashlyticsLoggerFactory

@Suppress("unused") // SLF4J API uses this class
class StaticLoggerBinder private constructor() : LoggerFactoryBinder {
    /* ====================================================================== */
    override fun getLoggerFactory(): ILoggerFactory {
        return factory
    }

    override fun getLoggerFactoryClassStr(): String {
        return factory.javaClass.name
    }

    companion object {
        private val factory = CrashlyticsLoggerFactory()

        @Suppress("unused") // SLF4J API uses this getter
        @get:JvmStatic
        val singleton = StaticLoggerBinder()
    }
}